testservices/neutronmodel/neutronmodel.go

```diff
@@ -320,8 +321,26 @@ func (n *NeutronModel) AddSecurityGroupRule(ruleId string, rule neutron.RuleInfo
 	if rule.IPProtocol != "" {
 		newrule.IPProtocol = &rule.IPProtocol
 	}
-	if rule.EthernetType != "" {
-		newrule.EthernetType = rule.EthernetType
+	switch rule.EthernetType {
+		case "":
+			// Neutron assumes IPv4 if no EthernetType is specified
+			newrule.EthernetType = "IPv4"
+			break
+		case "IPv4", "IPv6":
+			newrule.EthernetType = rule.EthernetType
+			break
```

Comment:
@fnordahl You shouldn't need the `break`-s. Go will effectively break after the matching case block, unless you use `fallthrough`.